sasl/aws_msk_iam_v2/msk_iam.go

```diff
@@ -164,3 +169,13 @@ func defaultSignTime(v time.Time) time.Time {
 	}
 	return v
 }
+
+// NewMechanism provides
+func NewMechanism(awsCfg aws.Config) *Mechanism {
+	sig := signer.NewSigner()
+	return &Mechanism{
+		Signer:              sig,
+		CredentialsProvider: awsCfg.Credentials,
+		Region:              awsCfg.Region,
+	}
```

Comment:
It seems we could do without the intermediary variable here?

```suggestion
	return &Mechanism{
		Signer:              signer.NewSigner(),
		CredentialsProvider: awsCfg.Credentials,
		Region:              awsCfg.Region,
	}
```